numpy/core/src/multiarray/scalartypes.c.src

```diff
@@ -1549,6 +1549,45 @@ gentype_@name@(PyObject *self, PyObject *args, PyObject *kwds)
 }
 /**end repeat**/
 
+static PyObject *
+gentype_dunder_round(PyObject *self, PyObject *args, PyObject *kwds)
+{
+    static char *kwlist[] = {"ndigits", NULL};
+    PyObject *ndigits = Py_None;
+    if (!PyArg_ParseTupleAndKeywords(args, kwds, "|O:__round__", kwlist, &ndigits)) {
+        return NULL;
+    }
+    
+    PyObject *tup;
+    if (ndigits == Py_None) {
+        tup = PyTuple_Pack(0);
+    }
+    else {
+        tup = PyTuple_Pack(1, ndigits);
+    }
+
+    if (tup == NULL) {
+        return NULL;
+    }
+
+    PyObject *obj = gentype_round(self, tup, NULL);
+    Py_DECREF(tup);
+    if (obj == NULL) {
+        return NULL;
+    }
+
+    PyObject *ret;
+    if (ndigits == Py_None) {
+        ret = PyNumber_Long(obj);
+    }
+    else {
+        return obj;
+    }
+    Py_DECREF(obj);
+
+    return ret;
```

Comment:
It would make more sense to me to either put these lines in the `if` above, or move the line within the if down to here.